frontend/src/Editor/Inspector/EventManager.jsx

```diff
@@ -603,62 +605,156 @@ export const EventManager = ({
     );
   }
 
-  function renderHandlers(events) {
-    return events.map((event, index) => {
-      const actionMeta = ActionTypes.find((action) => action.id === event.actionId);
-      const rowClassName = `row g-0 border-bottom pb-2 pt-2 px-2 ${focusedEventIndex === index ? ' bg-azure-lt' : ''}`;
-
-      return (
-        <div key={index}>
-          <OverlayTrigger
-            trigger="click"
-            placement={popoverPlacement || 'left'}
-            rootClose={true}
-            overlay={eventPopover(event, index)}
-            onHide={() => setFocusedEventIndex(null)}
-            onToggle={(showing) => {
-              if (showing) {
-                setFocusedEventIndex(index);
-              } else {
-                setFocusedEventIndex(null);
-              }
-              if (typeof popOverCallback === 'function') popOverCallback(showing);
-            }}
-          >
-            <div className="card mb-1">
-              <div className="card-body p-0" data-cy="event-handler-card">
-                <div className={rowClassName} role="button">
-                  <div className="col" data-cy="event-handler">
-                    {componentMeta.events[event.eventId]['displayName']}
-                  </div>
-                  <div className="col" data-cy="event-name">
-                    <small className="event-action font-weight-light">{actionMeta.name}</small>
-                  </div>
-                  <div className="col-auto">
-                    <span
-                      className="text-danger"
-                      onClick={(e) => {
-                        e.stopPropagation();
-                        removeHandler(index);
-                      }}
-                      data-cy="delete-button"
-                    >
-                      <svg width="10" height="16" viewBox="0 0 10 16" fill="none" xmlns="http://www.w3.org/2000/svg">
-                        <path
-                          d="M0 13.8333C0 14.75 0.75 15.5 1.66667 15.5H8.33333C9.25 15.5 10 14.75 10 13.8333V3.83333H0V13.8333ZM1.66667 5.5H8.33333V13.8333H1.66667V5.5ZM7.91667 1.33333L7.08333 0.5H2.91667L2.08333 1.33333H0V3H10V1.33333H7.91667Z"
-                          fill="#8092AC"
-                        />
-                      </svg>
-                    </span>
-                  </div>
-                </div>
-              </div>
+  const reorderEvents = (startIndex, endIndex) => {
+    const result = component.component.definition.events;
+    const [removed] = result.splice(startIndex, 1);
+    result.splice(endIndex, 0, removed);
+    eventsChanged(result);
```

Comment:
@kavinvenkatachalam, Even if I removed this `eventsChanged` call, the reordering function will still work, due to the mutation caused by the preceding two `splice` calls. Though I don't see any bugs, I'm feeling a bit scared about introducing state mutation here, can we do it without mutation? At worst we can just use 
`const result = _.cloneDeep(component.component.definition.events);`  on line 609

Perhaps we can do something similar on the third splice call that exists on the `removeHandler` function as well.